src/types.ts

```diff
@@ -412,6 +412,11 @@ export interface FunctionOptions {
      */
     period: TimeStr;
   };
+
+  /**
+   * Specifies the maximum number of retries for all steps across this function.
+   */
+  retries?: number;
```

Comment:
I can see someone passing in `Infinity` here to try to get it to never stop retrying.

Is a function with unlimited retries possible atm?

Reply:
Its not, it caps at 20

Reply:
Added some verbose validation to this for `0` to `20`. Should warn most users early, and then the executor can complain if it gets deployed with a number outside the range.